lib/App/MintTag.pm

```diff
@@ -493,6 +495,41 @@ sub _maybe_delete_source_branches ($self, $step) {
   }
 }
 
+sub _maybe_cleanup_tags ($self, $step) {
+  return unless $step->push_tag_to && $step->cleanup_tag_days;
+
+  my $remote = $step->push_tag_to;
+  my $prefix = $step->tag_prefix;
+
+  try {
+    my $tag_spec = "refs/tags/${prefix}*";
+    my $tag_list = run_git('for-each-ref', "--format=%(refname:short)\t%(creatordate:unix)", $tag_spec);
+
+    my @tags = map {; my ($tag, $ts) = split /\t/; { ts => $ts, 'tag' => $tag} }
+    split /\n/, $tag_list;
+
+    my $time = time;
+
+    my $max_ts = $time - ($step->cleanup_tag_days * 86400);
+
+    my @can_remove = map {; $_->{tag} } grep {; $_->{ts} < $max_ts } @tags;
+
+    while (my @batch = splice @can_remove, 0, 100) {
+      $Logger->log_debug(["Deleting tags: %s", join(' ', @batch)]);
+      run_git('push', $remote->name, map {; ":refs/tags/$_" } @batch);
+    }
+
+  } catch {
+    my $err = $_;
+
+    $Logger->log_fatal([
```

Comment:
log_fatal will throw an exception and maybe throw a wrench into the minting process. I'm not sure failing to clean up old refs is worth halting a mint in progress?